Sources/Datadog/RUM/RUMMonitor/Scopes/RUMSessionScope.swift

```diff
@@ -14,27 +14,59 @@ internal class RUMSessionScope: RUMScope {
         static let sessionMaxDuration: TimeInterval = 4 * 60 * 60 // 4 hours
     }
 
+    // MARK: - Child Scopes
+
+    /// Active View scopes. Scopes are added / removed when the View starts / stops displaying.
+    private(set) var viewScopes: [RUMScope] = []
+
     // MARK: - Initialization
 
     unowned let parent: RUMScope
```

Comment:
does `parent` need to be a `RUMScope`? if we split `RUMScope` protocol into `ContextProvider` and `CommandHandler` (i made up those names) then we can do `parent: ContextProvider`; that way we ensure commands are propagated top to down only.
does that make sense?

Reply:
Yes, it makes sense 👌. Indeed, the `parent` reference is to only get the `parent.context`. The `RUMContextProvider` interface is planned for `RUMM-597: Current `RUMContext` is available globally`. Let's try this refactoring then - we will have the knowledge on all scopes (right now few are still missing). WDYT?

Reply:
a `TODO` here would be 👌 then

Reply:
Added 👌 https://github.com/DataDog/dd-sdk-ios/pull/193/commits/fed1d347a78fb02e52a6af6419558371e405111c I also updated `RUMM-597` with this notice.